src/lib.rs

```diff
@@ -563,6 +575,228 @@ impl<'a> LabelText<'a> {
     }
 }
 
+
+/// This structure holds all information that can desribe an arrow connected to
+/// either start or end of an edge.
+#[derive(Clone, Hash, PartialEq, Eq)]
+pub struct Arrow {
+    pub arrows: Vec<ArrowShape>,
+}
+
+use self::ArrowShape::*;
+
+impl Arrow {
+
+    /// Arrow constructor which returns an empty arrow
+    pub fn none() -> Arrow {
+        Arrow {
+            arrows: vec![NoArrow],
+        }
+    }
+
+    /// Arrow constructor which returns a regular triangle arrow, without modifiers
+    pub fn normal() -> Arrow {
+        Arrow {
+            arrows: vec![Normal],
+        }
+    }
+
+    /// Arrow constructor which returns an arrow created by a given ArrowShape.
+    pub fn from_arrow(arrow: ArrowShape) -> Arrow {
+        Arrow {
+            arrows: vec![arrow],
+        }
+    }
+
+    /// Function which converts given arrow into a renderable form.
+    pub fn to_dot_string(&self) -> String {
+        let mut cow = String::new();
+        for arrow in &self.arrows {
+            cow.push_str(&arrow.to_dot_string());
+        };
+        cow
+    }
+}
+
+
+impl Into<Arrow> for [ArrowShape; 2] {
+    fn into(self) -> Arrow {
+        Arrow {
+            arrows: vec![self[0], self[1]],
+        }
+    }
+}
+impl Into<Arrow> for [ArrowShape; 3] {
+    fn into(self) -> Arrow {
+        Arrow {
+            arrows: vec![self[0], self[1], self[2]],
+        }
+    }
+}
+impl Into<Arrow> for [ArrowShape; 4] {
+    fn into(self) -> Arrow {
+        Arrow {
+            arrows: vec![self[0], self[1], self[2], self[3]],
+        }
+    }
+}
+
+/// Arrow modifier that determines if the shape is empty or filled.
+#[derive(Clone, Copy, Hash, PartialEq, Eq)]
+pub enum Fill {
+    Open,
+    Filled,
+}
+
+impl Fill {
+    pub fn as_slice(self) -> &'static str {
+        match self {
+            Fill::Open => "o",
+            Fill::Filled => "",
+        }
+    }
+}
+
+/// Arrow modifier that determines if the shape is clipped.
+/// For example `Side::Left` means only left side is visible.
+#[derive(Clone, Copy, Hash, PartialEq, Eq)]
+pub enum Side {
```

Comment:
Don't make the change just because I said so. I wasn't sure if you'd thought of that option. 

I'll merge it either way :-)

On 16 Nov 2015, at 8:45 AM, Ygg01 <notifications@github.com> wrote:
> 
> In src/lib.rs:
> 
> > +    Filled,
> > +}
> > +
> > +impl Fill {
> > +    pub fn as_slice(self) -> &'static str {
> > +        match self {
> > +            Fill::Open => "o",
> > +            Fill::Filled => "",
> > +        }
> > +    }
> > +}
> > +
> > +/// Arrow modifier that determines if the shape is clipped.
> > +/// For example `Side::Left` means only left side is visible.
> > +#[derive(Clone, Copy, Hash, PartialEq, Eq)]
> > +pub enum Side {
> Makes sense, I guess. I was kinda undecided whether to go for Both or Option<Side>. It made sense that it's presence or lack of a side, but if you think Both is more meaningful, then its better to do it that way.
> 
> —
> Reply to this email directly or view it on GitHub.
>